packages/sources/coinmetrics/src/endpoint/burned.ts

```diff
@@ -1,63 +1,164 @@
+
+  if (startDate && !reDate.test(startDate)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'startDate': ${startDate}. Expected formats is: YYYY-MM-DD.`,
+      statusCode: 400,
+    })
+  }
+
+  if (Date.parse(startDate) > Date.parse(today)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'startDate': ${startDate}. Is greater than today: ${today}.`,
+      statusCode: 400,
+    })
+  }
+
+  if (!reDate.test(endDate)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'endDate': ${endDate}. Expected formats is: YYYY-MM-DD.`,
+      statusCode: 400,
+    })
+  }
+
+  if (Date.parse(endDate) > Date.parse(today)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'endDate': ${endDate}. Is greater than today: ${today}.`,
+      statusCode: 400,
+    })
+  }
+
+  if (Date.parse(startDate) > Date.parse(endDate)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid date range: 'startDate': ${startDate} cannot be greater than 'endDate': ${endDate}.`,
+      statusCode: 400,
+    })
+  }
 
   const params = {
-    assets: (asset as string).toLowerCase(),
-    metrics,
-    frequency,
+    assets: ASSETS,
+    metrics: METRICS,
+    frequency: FREQUENCY,
+    page_size: PAGE_SIZE,
+    start_time: startDate,
+    end_time: endDate,
     api_key: config.apiKey,
-    page_size: pageSize,
   }
 
-  const options = { ...config.api, params, url }
+  const options = { ...config.api, params, url: URL }
+
+  let totalBurnedETH = new Decimal(0)
+  let response
+  /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
+  while (true) {
+    response = await Requester.request(options)
 
-  const response = await Requester.request<ResponseSchema>(options)
-  const FeeTotNtv = ethers.utils.parseEther(response.data.data[0].FeeTotNtv)
-  const RevNtv = ethers.utils.parseEther(response.data.data[0].RevNtv)
-  const IssTotNtv = ethers.utils.parseEther(response.data.data[0].IssTotNtv)
-  const result = FeeTotNtv.sub(RevNtv.sub(IssTotNtv))
+    const responseData: ResponseData = response.data
+    const { data: assetMetricsList } = responseData
+    if (!Array.isArray(assetMetricsList)) {
+      throw new AdapterError({
+        jobRunID,
+        message: `Unexpected response: ${JSON.stringify(
+          assetMetricsList,
+        )}. 'data' expected to be an array.`,
+        statusCode: 400,
+      })
+    }
+    totalBurnedETH = totalBurnedETH.plus(calculateBurnedETH(jobRunID, assetMetricsList))
+
+    const nextPageToken = response.data.next_page_token
+    if (!nextPageToken || assetMetricsList.length < PAGE_SIZE) break
+    options.params.next_page_token = nextPageToken
+  }
 
   return Requester.success(
     jobRunID,
-    Requester.withResult(response, ethers.utils.formatEther(result)),
+    Requester.withResult(
+      response,
+      ethers.utils.parseUnits(totalBurnedETH.toString(), 'ether').toString(),
```

Comment:
The data provider returns the units in ether, so we should also return the units in ether.